util/update_logdefs.ts

```diff
@@ -0,0 +1,374 @@
+  }
+
+  buildPullRequestBodyContent(): string {
+    let output = '';
+    for (const type of this.logDefsNoInclude) {
+      const triggerMatches = this.triggerMatches[type] ?? [];
+      const timelineMatches = this.timelineMatches[type] ?? [];
+      if (triggerMatches.length === 0 && timelineMatches.length === 0)
+        continue;
+      output += `\n## \`${type}\`\n`;
+      triggerMatches.forEach((m) => {
+        output += `${this.buildRefUrl(m.filename, m.line, sha, true)}\n`;
+      });
+      timelineMatches.forEach((m) => {
+        output += `${this.buildRefUrl(m.filename, m.line, sha, false)}\n`;
+      });
+    }
+    return output;
+  }
+
+  processAndLogResults(): void {
+    // log results to the console for both CLI & GH workflow execution
+    for (const type of this.logDefsNoInclude) {
+      const matches = [
+        ...this.triggerMatches[type] ?? [],
+        ...this.timelineMatches[type] ?? [],
+      ];
+      if (matches.length === 0)
+        continue;
+
+      console.log(`** ${type} **`);
+      console.log(`Found non-included log line type in active use:`);
+
+      matches.forEach((m) => {
+        console.log(`  - ${m.filename}:${m.line}`);
+      });
+
+      console.log(`LOG DEFS UPDATED: ${type} is being added to the analysis filter.\n`);
+      this.logDefsToUpdate.push(type);
+    }
+
+    // Log a notice for 'never' log line types, just so we're aware of the usage count for each.
+    // In theory, these are set to 'never' because we really don't care about them for analysis,
+    // but a periodic reminder to re-evaluate never hurts.
+    for (const type of this.logDefsNeverInclude) {
+      const numMatches = (this.triggerMatches[type]?.length ?? 0) +
+        (this.timelineMatches[type]?.length ?? 0);
+      if (numMatches > 0) {
+        console.log(`** ${type} **`);
+        console.log(
+          `Found ${numMatches} active use(s) of suppressed ('never') log line type.`,
+        );
+        console.log(
+          `${type} will not be added to the analysis filter, but please consider whether updates are needed.\n`,
+        );
+      }
+    }
+  }
+
+  getFileList(): FileList {
+    const fileList: FileList = {
+      timelines: [],
+      triggers: [],
+    };
+
+    walkDirSync(path.posix.join(this.projectRoot, raidbossRelDir), (filepath) => {
+      if (/\/raidboss_manifest.txt/.test(filepath)) {
+        return;
+      }
+      if (/\/raidboss\/data\/.*\.txt/.test(filepath)) {
+        fileList.timelines.push(filepath);
+        return;
+      }
+      if (/\/raidboss\/data\/.*\.[jt]s/.test(filepath)) {
+        fileList.triggers.push(filepath);
+        return;
+      }
+    });
+
+    return fileList;
+  }
+
+  parseTriggerFile(file: string): void {
+    // We could dynamically import each trigger file and get the type from triggerSet,
+    // but we would lose the line number, which we need to pass the URL to the PR body,
+    // so just use regex to parse the file instead.
```

Comment:
As always, I'm going to grumble here about parsing files with regex when another option is available.

Would it be possible to do something like:

1. Import the file normally
2. Loop through the `triggers` and `timelineTriggers`
3. If there's a trigger found, then regex the file for something like `id: '${trigger.id}'.*?(?:netRegex|regex): .*?$` to find the line number

Reply:
Yeah, I think that works.  I'd normally grumble about the cost of having to import and then loop-regex the file for each match, but the matches should be very small going forward, and I hate using regex to parse JS files (maybe not as much as you, but probably close).  Also, thanks -- that StackOverflow comment absolutely made my day. 🤣